### PR TITLE
Convert addons' info to addon.json

### PR DESCRIPTION
**Pull request by DaazKu**
Depends on 
- ~~https://github.com/vanilla/vanilla/pull/5545 must be merged for the unit tests to work properly. I will rebase afterward.~~
- ~~https://github.com/vanilla/vanilla/pull/5541~~

Note that I updated .gitignore with 2 missing plugins.

Fixes https://github.com/vanilla/vanilla/issues/5112

### CODE REVIEW

**Comment by initvector** [comment]
On hold pending a fix for the build

**Comment by DaazKu** [comment]
Unit tests wont pass until I rebase this from #5541

File: applications/conversations/settings/class.hooks.php [hunk 1]
```diff
@@ -197,9 +197,8 @@ public function setup() {
         include(PATH_APPLICATIONS.DS.'conversations'.DS.'settings'.DS.'structure.php');
         include(PATH_APPLICATIONS.DS.'conversations'.DS.'settings'.DS.'stub.php');
 
-        $ApplicationInfo = array();
-        include(combinePaths(array(PATH_APPLICATIONS.DS.'conversations'.DS.'settings'.DS.'about.php')));
-        $Version = val('Version', val('Conversations', $ApplicationInfo, array()), 'Undefined');
-        saveToConfig('Conversations.Version', $Version);
+        // Set current Conversations.Version
+        $appInfo = json_decode(file_get_contents(PATH_APPLICATIONS.DS.'conversations'.DS.'addon.json'), true);
+        saveToConfig('Conversations.Version', val('version', $appInfo, 'Undefined'));
```
Comment (initvector) [inline applications/conversations/settings/class.hooks.php]: This is only going to update the version on enable. What about when the files are copied over and the user runs /utility/update? The version, even if updated in addon.json, would stay the same.

Comment (DaazKu) [inline applications/conversations/settings/class.hooks.php]: I'm just updating the current behaviour...

**Pull request merged**